pkg/apis/pipeline/v1beta1/param_context.go

```diff
@@ -127,19 +131,26 @@ func getContextParams(ctx context.Context, overlays ...Param) []Param {
 
 		// If there is no overlay, pass through the param to the next level.
 		// e.g. for strings $(params.name), for arrays $(params.name[*]).
+		// for object: {key: param's name, value: $(params.name[*])}
 		p := Param{
 			Name: ps.Name,
 		}
-		if ps.Type == ParamTypeString {
+		switch ps.Type {
+		case ParamTypeString:
 			p.Value = ArrayOrString{
 				Type:      ParamTypeString,
 				StringVal: fmt.Sprintf("$(params.%s)", ps.Name),
 			}
-		} else {
+		case ParamTypeArray:
 			p.Value = ArrayOrString{
 				Type:     ParamTypeArray,
 				ArrayVal: []string{fmt.Sprintf("$(params.%s[*])", ps.Name)},
 			}
+		default:
+			p.Value = ArrayOrString{
+				Type:      ParamTypeObject,
+				ObjectVal: map[string]string{ps.Name: fmt.Sprintf("$(params.%s[*])", ps.Name)},
+			}
```

Comment:
I'm not sure why the old comment gets resolved, but we probably don't want to make the default to be setting the type to be ParamTypeObject. Please error this out. This is different from the other default case in another comment thread

Reply:
please follow up on the original thread.